keyboards/kbd6x/keymaps/othi/keymap.c

```diff
@@ -0,0 +1,225 @@
+    DE_OE, /* Ctrl + Shift + u00F6 */
+    DE_UE, /* Ctrl + Shift + u00FC */
+    FLUSH, /* clears oneshots */
+    DE_AE_CAP, /* Ctrl + Shift + u00C4 */
+    DE_OE_CAP, /* Ctrl + Shift + u00D6 */
+    DE_UE_CAP, /* Ctrl + Shift + u00DC */
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch(keycode) {
+            case MY_CUSTOM_MACRO:
+                SEND_STRING("QMK is the best thing ever!"); // this is our macro!
+                return false;
+            case DE_AE:
+                SEND_STRING(SS_DOWN(X_LCTRL) SS_DOWN(X_LSHIFT) SS_TAP(X_U) SS_UP(X_LCTRL) SS_UP(X_LSHIFT) SS_TAP(X_0) SS_TAP(X_0) SS_TAP(X_E) SS_TAP(X_4) SS_TAP(X_ENTER));
+                return false;
+            case DE_AE_CAP:
+                SEND_STRING(SS_DOWN(X_LCTRL) SS_DOWN(X_LSHIFT) SS_TAP(X_U) SS_UP(X_LCTRL) SS_UP(X_LSHIFT) SS_TAP(X_0) SS_TAP(X_0) SS_TAP(X_C) SS_TAP(X_4) SS_TAP(X_ENTER));
+                return false;
+            case DE_OE:
+                SEND_STRING(SS_DOWN(X_LCTRL) SS_DOWN(X_LSHIFT) SS_TAP(X_U) SS_UP(X_LCTRL) SS_UP(X_LSHIFT) SS_TAP(X_0) SS_TAP(X_0) SS_TAP(X_F) SS_TAP(X_6) SS_TAP(X_ENTER));
+                return false;
+            case DE_OE_CAP:
+                SEND_STRING(SS_DOWN(X_LCTRL) SS_DOWN(X_LSHIFT) SS_TAP(X_U) SS_UP(X_LCTRL) SS_UP(X_LSHIFT) SS_TAP(X_0) SS_TAP(X_0) SS_TAP(X_D) SS_TAP(X_6) SS_TAP(X_ENTER));
+                return false;
+            case DE_UE:
+                SEND_STRING(SS_DOWN(X_LCTRL) SS_DOWN(X_LSHIFT) SS_TAP(X_U) SS_UP(X_LCTRL) SS_UP(X_LSHIFT) SS_TAP(X_0) SS_TAP(X_0) SS_TAP(X_F) SS_TAP(X_C) SS_TAP(X_ENTER));
+                return false;
+            case DE_UE_CAP:
+                SEND_STRING(SS_DOWN(X_LCTRL) SS_DOWN(X_LSHIFT) SS_TAP(X_U) SS_UP(X_LCTRL) SS_UP(X_LSHIFT) SS_TAP(X_0) SS_TAP(X_0) SS_TAP(X_D) SS_TAP(X_C) SS_TAP(X_ENTER));
+                return false;
+            case DE_SS:
+                SEND_STRING(SS_DOWN(X_LCTRL) SS_DOWN(X_LSHIFT) SS_TAP(X_U) SS_UP(X_LCTRL) SS_UP(X_LSHIFT) SS_TAP(X_0) SS_TAP(X_0) SS_TAP(X_D) SS_TAP(X_F) SS_TAP(X_ENTER));
+                return false;
+        }
+    }
+    return true;
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
+    switch (biton32(state)) {
+    case NM_MODE:
+        rgblight_setrgb (0x00,  0x66, 0x00);
+        break;
+    case VI_MODE:
+        rgblight_setrgb (0x66,  0x66, 0x00);
+        break;
+    case ACCENT:
+        rgblight_setrgb (0x7A,  0x00, 0xFF);
+        break;
+    case ACCENT_CAP:
+        rgblight_setrgb (0x7A,  0xFF, 0xFF);
+        break;
+    default: //  for any other layers, or the default layer
+        rgblight_setrgb (0xFF,  0x00, 0x00);
+        break;
+    }
+  return state;
+}
+
+//Tap Dance Declarations
+enum {
+  CTRL_NM = 0,
+  ALT_NM = 1,
+  SFT_NM = 2,
+  GUI_NM = 3
+};
+
+void dance_CTRL_NM_finished (qk_tap_dance_state_t *state, void *user_data) {
+  if (state->count == 1) {
+	register_code (KC_LCTL);
```

Comment:
Depending on what you're doing, you may not need the `register_code` function here, as the one shot mod should send that with the next keypress. 

Also, if you want it for both counts, you could just move it to before the check.